aws/tags.go

```diff
@@ -93,14 +93,35 @@ func setVolumeTags(conn *ec2.EC2, d *schema.ResourceData) error {
 				if err != nil {
 					ec2err, ok := err.(awserr.Error)
 					if ok && strings.Contains(ec2err.Code(), ".NotFound") {
-						return resource.RetryableError(err) // retry
+						return resource.RetryableError(err)
 					}
 					return resource.NonRetryableError(err)
 				}
 				return nil
 			})
 			if err != nil {
-				return err
+				if !isResourceTimeoutError(err) && !isAWSErr(err, "InvalidParameterValueException", "Your request has been throttled by EC2") {
```

Comment:
I'm really sorry I didn't provide a better example for here. 😅 I believe our new goal here was to remove the time bounding caused by these `resource.Retry()` functions and instead allow the SDK to control retries if the timeout of the existing `resource.Retry()` loop didn't successfully complete.

To that end, I believe what we are looking for here is simply the following for each of these cases:

```go
}) // end of existing resource.Retry()

// Retry without time bounds for EC2 throttling
if isResourceTimeoutError(err) {
  _, err = conn.DeleteTags(input)
}

if err != nil {
  // existing error logic
```

Rinse and repeat for the others. Again sorry for any confusion or if I'm missing something here.

Reply:
oh I see where you want to go with this. Yeah from the example you gave I thought you wanted to add extra slack if the initial timeout was due to throttling. this makes it much easier and less bloated. I will get to it asap 👍